subsys/bluetooth/controller/hci_driver.c

```diff
@@ -328,6 +328,22 @@ static inline void receive_signal_raise(void)
 	mpsl_work_submit(&receive_work);
 }
 
+static struct {
+	uint8_t buf[HCI_RX_BUF_SIZE];
+	sdc_hci_msg_type_t type;
+	bool fetched;
+} rx_hci_msg;
```

Comment:
Suggest to add some more documentation to make it more clear how this is used.
```suggestion
/** Storage for HCI packets from controller to host */
static struct {
	uint8_t buf[HCI_RX_BUF_SIZE];
	/** The type of HCI packet the buffer contains. */
	sdc_hci_msg_type_t type;
	/** Set to true if the buffer contains a packet*/
	bool fetched;
} rx_hci_msg;
```

Edit: When looking at it a bit more, we could benefit from defining `SDC_HCI_MSG_TYPE_NONE`, then we would not need the additional bool.